src/lean_dojo/data_extraction/lean.py

```diff
@@ -455,7 +457,7 @@ def is_lean4(self) -> bool:
 
     @property
     def commit_url(self) -> str:
-        return os.path.join(self.url, f"tree/{self.commit}")
+        return f"{self.url}/tree/{self.commit}"
```

Comment:
What if `self.url` ends with `/`?

Reply:
good question. It is striped here:
```py
object.__setattr__(self, "url", normalize_url(self.url))
```

where
```py
def normalize_url(url: str) -> str:
    return _URL_REGEX.fullmatch(url)["url"]  # Remove trailing `/`.
```